docs/formatter.qmd

```diff
@@ -174,6 +174,298 @@ list(foo, bar)
 
 The goal of this feature is to strike a balance between being opinionated and recognizing that users often know when taking up more vertical space results in more readable output.
 
+# Autobracing
+
+To encourage more consistent, readable, and portable code, Air will *autobrace* the following elements:
+
+-   If statements
+
+-   For, while, and repeat loops
+
+-   Function definitions
+
+Autobracing is the process of wrapping the body of these code elements with `{ }` if braces don't already exist.
+
+## If statements
+
+Air will autobrace if statements if:
+
+-   Any existing part of the if statement spans multiple lines
+
+-   Any existing part of the if statement is already braced
+
+-   The if statement is nested, i.e. there is an `else if {`
+
+-   The if statement exceeds the line length
+
+For example, the following will all be autobraced:
+
+``` r
+if (condition)
+  a
+
+# Becomes:
+if (condition) {
+  a
+}
+```
+
+``` r
+if (condition) a else { b }
+
+# Becomes:
+if (condition) {
+  a
+} else {
+  b
+}
+```
+
+``` r
+if (condition) a else if (condition2) b else c
+
+# Becomes:
+if (condition) {
+  a
+} else if (condition2) {
+  b
+} else {
+  c
+}
+```
+
+Simple if statements that don't hit any of the autobracing criteria mentioned above are allowed to stay on one line as long as they are also in *value* position, as opposed to *effect* position.
+
+-   Top level if statements are in effect position.
+
+-   If statements that are direct children of `{}` are in effect position, unless the if statement is the last child of the `{}` expression list, in which case it is in value position (because it is the returned value from that scope).
+
+-   Otherwise, the if statement is in value position.
+
+These if statements are in effect position, and would be autobraced:
+
+``` r
+# If statement at top level
+if (condition) a else b
+
+# If statement inside `{}`
+fn <- function() {
+  if (condition) stop("oh no")
+  1 + 1
+}
+
+fn <- function() {
+  if (condition) return(1)
+  1 + 1
+}
+```
```

Comment:
I would show the resulting formatting and mention that having side effects like stop or return on their own line is important for readability as they impact control flow or state. You could also add a `<-` example.